site/src/pages/api/verifyEmail.api.ts

```diff
@@ -0,0 +1,81 @@
+import { body as bodyValidator, validationResult } from "express-validator";
+import { createBaseHandler } from "../../lib/handler/baseHandler";
+import { SerializedUser, User } from "../../lib/model/user.model";
+import { formatErrors } from "../../util/api";
+
+export type ApiVerifyEmailRequestBody = {
+  userId: string;
+  emailVerificationCodeId: string;
+  code: string;
+};
+
+export type ApiVerifyEmailResponse = {
+  user: SerializedUser;
+};
+
+export default createBaseHandler<
+  ApiVerifyEmailRequestBody,
+  ApiVerifyEmailResponse
+>()
+  .use(
+    bodyValidator("userId").isString().notEmpty(),
+    bodyValidator("emailVerificationCodeId").isString().notEmpty(),
+    bodyValidator("code").isString().notEmpty(),
+  )
+  .post(async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json(formatErrors(...errors.array()));
+    }
+
+    const { db, body } = req;
+    const { userId } = body;
+
+    const user = await User.getById(db, { userId });
+
+    if (!user) {
+      return res.status(404).json(
+        formatErrors({
+          msg: "Could not find user with the provided id",
+          value: userId,
```

Comment:
We don't include `param: "userId"` here like we do in the other error messages - I'm assuming something to do with how the FE will use them?

Reply:
I simply forgot to add it, will do so. I'm yet sure whether the errors displayed on the frontend should be defined in the backend or not.